rasa/shared/core/training_data/story_reader/yaml_story_reader.py

```diff
@@ -302,8 +302,14 @@ def _parse_user_utterance(self, step: Dict[Text, Any]) -> None:
             self.current_step_builder.add_user_messages([utterance])
 
     def _validate_that_utterance_is_in_domain(self, utterance: UserUttered) -> None:
+
         intent_name = utterance.intent.get(INTENT_NAME_KEY)
 
+        # check if this is a retrieval intent
+        # in this case check only for the base intent in domain
+        if INTENT_MESSAGE_PREFIX in intent_name:
```

Comment:
The same comment with `INTENT_MESSAGE_PREFIX` -> `RESPONSE_IDENTIFIER_DELIMITER` and using `Message.separate_intent_response_key()` instead.